frontend/README.md

```diff
@@ -0,0 +1,389 @@
+- Includes several tools and APIs that you wont need unless you have a specific use case, it might be considered overkill for some apps.
+- It includes its own custom way when creating I/O operations, in case you have generic logic you want to reuse you'll have to migrate them.
+
+### React-Query
+
+React query is an I/O abstraction that allows you create custom logic to connect to an API while also provides its own state management.
+
+**Pros:**
+
+- Allows you to decouple the actual API call from the React code, having your generic gateways doing the job.
+- Native support for queries, mutations, paginated queries and more.
+- Includes a tagging system for cache, so you can decide when to refetch certain queries.
+
+**Cons:**
+
+- It doesn't include a custom state management.
+- It still uses hooks a primary API.
+
+### Decision
+
+Having worked on big enterprise level React apps and at the same time on smaller code bases, there are some compromises that you need to do based on the team, state of the business and overall development experience.
+
+I believe that having overkill startegies when is not necessary can be as harmful as not having them when its necessary as it can slow the development process by adding extra overhead or quirks just to support something that can be simply done by just using React.
+
+In this case the best combination for a small app like this, I believe that a combination of React's core state management system like hooks, providers, context, etc. Combined with a simple but powerful I/O tool like `react-query` should be the best fit.
+
+As `react-query` allows you control the actual API request, you can create base gateways and custom responses (models) to use across the app, and the `providers` should be able to spread the information by custom hooks.
+
+### State Data Structure
+
+As `react-query` will do the heavy lifting to keep the image state and information in memory. The thing that we need to keep track will be the list of `in-progress` uploads. In this case we'll be using a native React [reducer](https://beta.reactjs.org/reference/react/useReducer), where we are going to store a key value map for the uploads, it should look as the following:
+
+```json
+{
+ "upload-id":  UploadModel,
+ "upload-id-2":  UploadModel,
+}
+```
+
+And accessing the store to write should be done by dispatching changes to `add, delete and update`.
+
+## Design-System
+
+To ensure component reusability and avoid recreating the same components over and over, creating an set of initial `design-system` component which should provide the basic styles for commonly used controls like:
+
+1. Buttons
+2. Text
+3. Modals
+4. Colors
+
+### Style-Components
+
+The styled-component library becomes very handy when it comes to create extendible components based on a specific set of styles, it contains a `ThemeProvider` tha can be used to store the overall styling rules, like spacing, colors, fonts, etc.
```

Comment:
```suggestion
The styled-component library becomes very handy when it comes to creating extendable components based on a specific set of styles, it contains a `ThemeProvider` that can be used to store the overall styling rules, like spacing, colors, fonts, etc.
```
Sometimes you're putting the library names (styled-component, react-query, etc.) in backticks and sometimes you're not, I would just suggest you do a pass for just the names and make sure you're treating them consistently.